src/libs/actions/Policy.ts

```diff
@@ -2503,6 +2547,78 @@ function clearCategoryErrors(policyID: string, categoryName: string) {
     });
 }
 
+/**
+ * Accept user join request to a workspace
+ */
+function acceptJoinRequest(reportId: string, accountID: string, adminsRoomMessageReportActionID: string) {
+    const choice = 'accept';
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportId}`,
+            value: {
+                [adminsRoomMessageReportActionID]: {
+                    originalMessage: {choice},
+                },
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportId}`,
+            value: {
+                [adminsRoomMessageReportActionID]: {
+                    originalMessage: {choice: ''},
+                },
+            },
+        },
+    ];
+
+    const parameters = {
+        requests: JSON.stringify({[reportId]: {requests: [{accountID, adminsRoomMessageReportActionID}]}}),
+    };
+
+    API.write(WRITE_COMMANDS.ACCEPT_JOIN_REQUEST, parameters, {optimisticData, failureData});
+}
+
+/**
+ * Decline user join request to a workspace
+ */
+function declineJoinRequest(reportId: string, accountID: string, adminsRoomMessageReportActionID: string) {
+    const choice = 'decline';
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportId}`,
+            value: {
+                [adminsRoomMessageReportActionID]: {
+                    originalMessage: {choice},
+                },
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportId}`,
+            value: {
+                [adminsRoomMessageReportActionID]: {
+                    originalMessage: {choice: ''},
+                },
+            },
+        },
+    ];
+
+    const parameters = {
+        requests: JSON.stringify({[reportId]: {requests: [{accountID, adminsRoomMessageReportActionID}]}}),
+    };
+
+    API.write(WRITE_COMMANDS.DECLINE_JOIN_REQUEST, parameters, {optimisticData, failureData});
```

Comment:
If this is [patternB](https://github.com/Expensify/App/blob/main/contributingGuides/OFFLINE_UX.md#b---optimistic-with-feedback-pattern) we need successData too, right?

Reply:
Good point, we should add a pendingAction: Update as well